runtime/vm/guardedstorage.c

```diff
@@ -28,7 +28,6 @@
 
 #if defined(OMR_GC_CONCURRENT_SCAVENGER) && defined(J9VM_ARCH_S390)
 
-J9_EXTERN_BUILDER_SYMBOL(handleSoftwareReadBarrier);
 J9_EXTERN_BUILDER_SYMBOL(handleHardwareReadBarrier);
```

Comment:
Moving the hardware read barrier turns out to be more difficult. The existing `handleHardwareReadBarrier` is a VM function and get used by VM function in `guardedStorage.c`. So nothing difficult is there because all are in the same module.

If it's moved to the JIT `znathelp.m4` and gets used in `j9gs_params_init` (a port lib function), we have to build the port lib against the JIT lib. Doing this is OK if we just build a JVM. But it'll be problematic if we are to build BC verify unit test in `openJ9/runtime/tests/bcvunit`. This test loads the port lib to perform tests. If the port lib gets an extra dependency on the JIT, the makefile of this needs changes too.

I would prefer to have this in a separate pull request as this PR is only for SW CS.

Reply:
@NigelYiboYu can you link the issue here for reference? I'd like to subscribe.

Reply:
The issue is #5620 for reference.